### DISCUSSION TITLE
Cannot search using searchable when use format

### DISCUSSION THREAD

**Discussion by SupunSpera** [rappasoft/laravel-livewire-tables#552] [Q&A]
Column::make('Referrer')->searchable()
                ->format(function ($created_at, $column, $customer) {
                    return $customer->referrer_name ;
                }),

cannot search using this searchable function when use format dunction. How can i do with rappasoft datatable?

**Comment by rizkhal** [comment]
My answer in you issue [here](https://github.com/rappasoft/laravel-livewire-tables/issues/553#issuecomment-972498836)

**Comment by anglin-rogersoft** [comment]
Column::make("Course number", "course_number")->sortable()->format(fn($value, $row, Column $column) => 'RGCS-'.$row->course_number)->searchable(),

I casted a column value as above. It's worked. But it's not searchable 
like rgc..

Is there any work arounds  guys ?

**Reply by lrljoe** [reply]
Yep, absolutley, you just need to use a callback for the searchable() method on the column 

Please see the docs on the rappasoft.com site for details

**Reply by anglin-rogersoft** [reply]
Thanks for the reply @lrljoe .

I tried as per #553  But not worked.

**Reply by lrljoe** [reply]
Hey @anglin-rogersoft - are you still having an issue with this?

If so, please can you share the code for your Column, and I'll give you some pointers.

**Reply by anglin-rogersoft** [reply]
Sorry @lrljoe , I was busy with some other works & forgot to update the status here. 

I got a solution.

Firstly I casted the column with model's getter.
`    public function getCourseNumberAttribute($courseNumber)
    {        
        return 'RGCS-'.$courseNumber;
    }`

Then I concatenate the prefix string with column in query builder.
` Column::make("Course number", "course_number")->sortable()->searchable(
                function (Builder $query, $term){
                    return $query->orWhere(DB::raw("CONCAT('RGCS-',course_number)"), 'like', '%' . trim($term) . '%');
                }
            ),`

However thank you so much @lrljoe for your responses, Kudos @rappasoft 🤗.

**Reply by lrljoe** [reply]
No problem, I do have a couple of items on the back burner for allowing improved sanitising of  search and filter inputs, so keep an eye out.

Try a label instead of a format in the interim

Use setAdditionalSelects to pull the value in your preferred style, or the builder() approach to include the attribute.

This will avoid the need to use any db::raw
